R/utils.R

```diff
@@ -342,14 +342,31 @@ beautify <- function(x, standardized = FALSE, use_math = FALSE, ...) {
 #' sanitize_terms(c("(Intercept)", "Factor A", "Factor B", "Factor A:Factor B", "scale(FactorA)"))
 #' }
 
-sanitize_terms <- function(x, standardized = FALSE) {
+sanitize_terms <- function(x, ...) {
+  UseMethod("sanitize_terms", x)
+}
+
+#' @rdname sanitize_terms
+#' @method sanitize_terms character
+
+sanitize_terms.character <- function(x, standardized = FALSE) {
   if(standardized) x <- gsub("scale\\(", "z_", x)   # Remove scale()
   x <- gsub("\\(|\\)|`", "", x)                     # Remove parentheses and backticks
+  x <- gsub("\\.0+$", "", x)                        # Remove trailing 0-digits
+  x <- gsub(",", "", x)                             # Remove big mark
   x <- gsub("\\W", "_", x)                          # Replace non-word characters with "_"
   x
 }
 
 
+#' @rdname sanitize_terms
+#' @method sanitize_terms data.frame
+
+sanitize_terms.data.frame <- function(x, ...) {
+  as.data.frame(lapply(x, sanitize_terms, ...))
```

Comment:
For old versions of R, we should add `stringsAsFactors = FALSE`. Moreover, `check.names = FALSE` would also make sense here (because it is an internal workhorse function and we definitely want to keep character vectors).